src/components/ShowService/ShowService.jsx

```diff
@@ -1,24 +1,45 @@
 import React, { Component } from 'react';
 import RenderService from '../ShowService/RenderService/RenderService';
+import { serviceService } from '../../services/service/service.service';
+import { Redirect } from 'react-router';
 
 class ShowService extends Component {
 	constructor(props) {
 		super(props);
 		this.state = {
-			service: []
+			service: {},
+			categories: [],
+			tutor: {},
+			tutorInfo: {},
+			redirectToTutor: false
 		};
+		this.setRedirect = this.setRedirect.bind(this);
+	}
+
+	
+	setRedirect () {
+		this.setState({
+			redirectToTutor: true
+		});
 	}
 
 	componentDidMount() {
-		fetch(`http://${process.env.REACT_APP_API_URI}:${process.env.REACT_APP_API_PORT}/api/services/${this.props.match.params.id}`)
-			.then(res => res.json())
+		fetch(serviceService.getDetailedById(this.props.match.params.id)
 			.then((data) => {
 				this.setState({ service: data });
-			})
+				this.setState({categories: data.categories});
+				this.setState({tutor: data.tutorInfo.user});
+				this.setState({tutorInfo: data.tutorInfo});
+				this.setState({categories: data.categories});
+			}))
```

Comment:
Det her skal være en setState. Så:
this.setState({
	service: data,
	categories: data.categories,
	tutor: data.tutorInfo.user,
	tutorInfo: data.tutorInfo,
	categories: data.categories
})

udover det indeholder din tutorInfo faktisk også tutor, så det giver ikke så meget mening at have en enkstra property til det.

Reply:
teknisk set indeholder service caetgories, tutorinfo og tutor, men de er med som subobject og dem kan man ikke tilgå medmindre man explicit lavet dem til object i tilstanden, ellers er de undefined :(